---
title: "Finish Aligning and Generate Read Counts"
author: "Yuchen Sun"
date: "2023-03-18"
output:
  html_document:
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Script repository location

***1. Set up a github repository where you will store all your scripts, and ideally, even your report at one point. Submit the URL of your repo.***

All project related scripts and documents are located in my ANGSD github repository [kevinsunofficial/angsd_project](https://github.com/kevinsunofficial/angsd_project). They are also located in my aphrodite directory `/home/yus4008/cmpb5004/project/angsd_project/`. For both repositories:

- `download.sh` - parse the SRA table and download the dataset from ENA
- `fastqc_trim.sh` - run fastqc analysis before and after the reads have been trimmed by trim-galore
- `create_index.sh` - create hg_38STARindex for STAR alignment
- `align.sh` - perform STAR alignment with created index and trimmed reads
- `featureCounts.sh` - perform featureCounts and create read counts for all samples (trimmed)
- `bamqc.sh` - run bamqc and multiqc on the alignment results

### Align all samples

***2. Align all your samples. Ideally use a for-loop within a script, i.e., automate and standardize the task to a certain extent, but do remember that legibility is valuable, too.***

The scripts for performing STAR alignment for all samples are located in `align.sh` as mentioned above. The pseudocode for the major logistic behind the script is described here:

```{pseudocode_align_all_samples,warning=F}
for response in uninfected symptomatic
do
    for file in response/paired_read1
    do
        get all information
        if not aligned already
        then
            STAR --runMode alignReads --readFilesIn paired_read1 paired_read2
            samtools index out_file
        fi
        if flagstat not exist
        then
            samtools flagstat out_file > out_flagstat
        fi
        if outstat not exist
        then
            samtools stats out_file > out_stats
        fi
    done
done
```

### Generate read count table

***3. Generate a read count table.***

The scripts for performing STAR alignment for all samples are located in `featureCounts.sh` as mentioned above. The pseudocode for the major logistic behind the script is described here:

```{pseudocode_feature_counts,warning=F}
featureCounts -p --countReadPairs -a gtf_file -t exon -g gene_id -o out_dir all_bam_files
```

### Load read counts

***4. Load the read count table into R and perform the quality controls and processing steps that we discussed in class.***

*`scp` `feature_counts.txt` and `feature_counts.txt.summary` from `/athena/angsd/scratch/yus4008/project/dataset/featureCounts/`.*

```{r environment_setup,warning=F,message=F}
library(dplyr)
library(tidyverse)
library(ggplot2)
library(DESeq2)
library(magrittr)
wd <- "C:/Yuchen/WCM/Courses/2023Spring/CMPB5004/project/angsd_project/fc_result/"
```

#### Load read count table

```{r load_read_count_table}
# load read counts and format the data table
readcounts <- read.table(paste0(wd,"feature_counts.txt"),header=T)
orig_names <- names(readcounts)
# preserve the SRR ID only
names(readcounts) <- gsub(".*(symptomatic|uninfected).(SRR)([0-9]{8}).*","\\2\\3",orig_names)
str(readcounts)
```

#### Make and filter DESeq2 dataset

```{r make_deseq2_count_col_data}
# generate the countData and colData for DESeq2 from the formatted read count
row.names(readcounts) <- make.names(readcounts$Geneid)
readcounts <- readcounts[,-c(1:6)]
head(readcounts)
sample_info <- data.frame(condition=gsub(".*(symptomatic|uninfected).*","\\1",orig_names)[-c(1:6)],
                          row.names=names(readcounts))
sample_info
```

```{r make_and_filter_deseq_dataset, warning=F}
# make the DESeq dataset from formatted countData and colData
DESeq.ds <- DESeqDataSetFromMatrix(countData=as.matrix(readcounts),colData=sample_info,design=~condition)
DESeq.ds
head(counts(DESeq.ds))
colSums(counts(DESeq.ds))
dim(DESeq.ds)

# remove genes with 0 reads
keep_genes <- rowSums(counts(DESeq.ds))>0
DESeq.ds <- DESeq.ds[keep_genes,]
dim(DESeq.ds)
```

#### Process DESeq2 dataset with log normalization and PCA

```{r log_normalized_counts}
DESeq.ds <- estimateSizeFactors(DESeq.ds)
assay(DESeq.ds,"log.norm.counts") <- log2(counts(DESeq.ds,normalized=T)+1)
```

```{r run_pca_on_log_norm_counts}
log.norm.counts = assay(DESeq.ds,"log.norm.counts")
pca <- prcomp(log.norm.counts,scale=T)
head(pca$rotation[,1:5])
qplot(c(1:length(pca$sdev)),pca$sdev^2/sum(pca$sdev^2)*100) + geom_line() +
  labs(x="PC",y="Variance Explained",main="Scree Plot") + ylim(0,100)
```

#### Run differential expression analysis

```{r prepare_run_DE_analysis}
DESeq.ds$condition %<>% relevel(ref="uninfected")
DESeq.ds %<>% DESeq()
```

```{r run_DE_analysis}
rowData(DESeq.ds) %>% colnames
DGE.results <- results(DESeq.ds,independentFiltering=T,alpha=0.05)
```


