---
title: "Download and Processing Data"
author: "Yuchen Sun"
date: "2023-02-17"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

***1. Download at least one FASTQ file that you will be working with for your project. Document the following details:***

```{download_from_ENA}
wget -O $fastqfile_output ftp://$ftplink
```

* ***where did you get it from?***
  + The data description metadata is retrieved from [SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP327322&f=immune_response_sam_s%3An%3Asymptomatic%2Cuninfected%3Ac&o=acc_s%3Aa), whereas the actual sequencing data (`.fastq.gz` files) is retrieved from [ENA](https://www.ebi.ac.uk/ena/browser/view/PRJNA744408).

* ***what publication is it linked to?***
  + The data is linked to [Transcriptome Analysis of Peripheral Blood Mononuclear Cells Reveals Distinct Immune Response in Asymptomatic and Re-Detectable Positive COVID-19 Patients](https://www.frontiersin.org/articles/10.3389/fimmu.2021.716075/full).

* ***who generated the data?***
  + The data is generated by Zhang J, Lin D, Li K, Ding X et al.

* ***how was the RNA extracted?***
  + The RNA extraction is done with TRIzol.

* ***what library prep was used?***
  + The library preparation is done with poly-A selection

* ***what cell type was used?***
  + The cell type used is peripheral blood mononuclear cells (PBMC).

* ***what was the treatment/experimental condition?***
  + The experimental condition is the symptomatic COVID-19 patients.

* ***what sequencing platform was used?***
  + The sequencing platform used is Illumina NovaSeq 6000 (Homo sapiens), with GEO [GPL24676](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL24676)



***2. Align the FASTQ file with an appropriate aligner (you may have to build a new index). Document:***

* ***parameters (and why you chose them)***

* ***summary of outcome and basic QC***