---
title: "Download and Processing Data"
author: "Yuchen Sun"
date: "2023-02-17"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

***1. Download at least one FASTQ file that you will be working with for your project. Document the following details:***

```{download_from_ENA}
wget -O $fastqfile_output ftp://$ftplink
```

* ***where did you get it from?***
  + The data description metadata is retrieved from [SRA Run Selector](https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP327322&f=immune_response_sam_s%3An%3Asymptomatic%2Cuninfected%3Ac&o=acc_s%3Aa), whereas the actual sequencing data (`.fastq.gz` files) is retrieved from [ENA](https://www.ebi.ac.uk/ena/browser/view/PRJNA744408).

* ***what publication is it linked to?***
  + The data is linked to [Transcriptome Analysis of Peripheral Blood Mononuclear Cells Reveals Distinct Immune Response in Asymptomatic and Re-Detectable Positive COVID-19 Patients](https://www.frontiersin.org/articles/10.3389/fimmu.2021.716075/full).

* ***who generated the data?***
  + The data is generated by Zhang J, Lin D, Li K, Ding X et al.

* ***how was the RNA extracted?***
  + The RNA extraction is done with TRIzol.

* ***what library prep was used?***
  + The library preparation is done with poly-A selection

* ***what cell type was used?***
  + The cell type used is peripheral blood mononuclear cells (PBMC).

* ***what was the treatment/experimental condition?***
  + The experimental condition is the symptomatic COVID-19 patients.

* ***what sequencing platform was used?***
  + The sequencing platform used is Illumina NovaSeq 6000 (Homo sapiens), with GEO [GPL24676](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GPL24676)



***2. Align the FASTQ file with an appropriate aligner (you may have to build a new index). Document:***

* ***parameters (and why you chose them)***
  + I used trim-galore to trim the reads. I used parameter of `--stringency 13` because the trim analysis suggests that with stringency being 13, it is more likely to prevent potentially wrong adapter trimming due to small overlaps.
  + I used STAR aligner. I first created an index for STAR with `--sjdbOverhang 99`. The reason I didn't use read length -1 is that I did trim-galore on the reads, therefore the trimmed reads are not uniformly with read length of 150bp. In the STAR manual it suggests that in most cases, the default value of 100 as read length will work well, therefore I chose 100-1=99 for this parameter.
  + I used `--outSAMtype BAM` to export the STAR alignment result in `.bam` format because it takes less space than `.sam`. I also added `SortedByCoordinate` so the information in the `.bam` file is sorted by coordinate.

* ***summary of outcome and basic QC***
  + The alignments have good mapping quality distributions, as around 70% of the reads are mapped with a MAPQ value higher than 241.